File: onyx-database/src/main/java/com/onyx/persistence/annotations/PostRemove.java
package com.onyx.persistence.annotations;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * This annotation is used to indicate method that is invoked after deleting a ManagedEntity.
 *
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * <pre>
 *     <code>
 *      {@literal @}PostRemove
 *      public void onPostRemove()
 *      {
 *          persistedObjects --;
 *      }
 *
 *     </code>
 * </pre>
 *
 * @see com.onyx.persistence.ManagedEntity
 *
 */
@SuppressWarnings("unused")
@Target({METHOD})
@Retention(RUNTIME)
public @interface PostRemove {
}

File: onyx-database/src/main/kotlin/com/onyx/persistence/annotations/PostRemove.kt
package com.onyx.persistence.annotations

/**
 * This annotation is used to indicate method that is invoked after deleting a ManagedEntity.
 *
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * @PostRemove
 * public void onPostRemove()
 * {
 *      persistedObjects --;
 * }
 *
 * @see com.onyx.persistence.ManagedEntity
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class PostRemove
